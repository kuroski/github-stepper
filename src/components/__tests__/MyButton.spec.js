import { render, fireEvent } from "@testing-library/vue";
import MyButton from "@/components/MyButton";

describe("MyButton", () => {
  it("works", async () => {
    const { getByText, emitted } = render(MyButton, {
      slots: {
        default: "Hello World"
      }
    });

    const button = getByText("Hello World");
    await fireEvent.click(button);

    expect(button).toMatchSnapshot();
    expect(emitted("click")).toMatchSnapshot();
  });
});
