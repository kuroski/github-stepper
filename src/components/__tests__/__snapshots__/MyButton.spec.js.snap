// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton works 1`] = `
<button
  class="button is-primary"
>
  Hello World
</button>
`;

exports[`MyButton works 2`] = `
Object {
  "click": Array [
    Array [
      "Hello button clicked",
    ],
  ],
}
`;
